add smbg.subType validation

jebeck opened the pull request:
What it says on the tin.

Review @kentquirk or @jh-bate? (And there's also a similar small PR open for ketones?)

jh-bate commented on test/schema/smbgTest.js:
```diff
@@ -43,5 +43,16 @@ describe('schema/cbg.js', function(){
+  describe('subType', function(){
+    var withSubType = _.assign({}, goodObject, {subType: ''});
+    helper.okIfAbsent(goodObject, 'subType');
+    helper.expectStringField(withSubType, 'subType');
```
am i missing something? isn't this just testing twice that it will pass muster if the subType = ''

jebeck replied:
No, the first one checks if it's OK that the whole field is absent. Note that I give it the unmodified `goodObject` there, not the new `withSubType` copy. Could split it into two tests if you like?

jh-bate replied:
I guess I was just expecting to see a test that had the subType as
either "linked" or "manual"

On Sat, Jan 31, 2015 at 6:52 AM, Jana Beck <notifications@github.com> wrote:

> In test/schema/smbgTest.js
> <https://github.com/tidepool-org/jellyfish/pull/48#discussion_r23858678>:
>
> > @@ -43,5 +43,16 @@ describe('schema/cbg.js', function(){
> >      helper.expectUnitConversion(goodObject, 'value');
> >    });
> >
> > +  describe('subType', function(){
> > +    var withSubType = _.assign({}, goodObject, {subType: ''});
> > +    helper.okIfAbsent(goodObject, 'subType');
> > +    helper.expectStringField(withSubType, 'subType');
>
> No, the first one checks if it's OK that the whole field is absent. Note
> that I give it the unmodified goodObject there, not the new withSubType
> copy. Could split it into two tests if you like?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tidepool-org/jellyfish/pull/48/files#r23858678>.
>

jh-bate commented:
I had a query on the tests other than that LGTM

jebeck commented:
Just added the subType to the smbg api ingestion test. There's gonna be a merge conflict with the ketones PR because I prettified the JSON for the api ingestion tests in the ketones PR. Whatever order these come in, someone just ping me and I'll clean up the mess of the conflicts.

kentquirk commented:
@jebeck -- please clean up. :)

jebeck commented:
@kentquirk squeaky clean :)

It turns out there also *was* a helper for testing specific values in the schemas, so I adjusted the test as well. I think that means I also addressed all your concerns re: tests @jh-bate, so this is ready for merge.
